### PR TITLE
Fix FuncTickFormatter.from_py_func missing return value and update examples

### LINKED ISSUE #5404: FuncTickFormatter.from_py_func() example ValueError
I am trying the FuncTickFormatter.from_py_func() example and I get the error
`ValueError: Function 'func' may only contain keyword arguments.`

A MWE that produces this error is:
```
from bokeh.models import FuncTickFormatter
from bokeh.plotting import figure, show
import math

def ticker(tick):
    return math.floor(tick) + " + " + (tick % 1).toFixed(2)

p = figure(plot_width=400, plot_height=300)
         
p.yaxis.formatter = FuncTickFormatter.from_py_func(ticker)

p.circle([2, 4, 6], [1, 3, 5])

show(p)
```

I am running Bokeh version 0.12.3 and have flexx installed.

### PR DESCRIPTION
**Pull request by canavandl**
Fix FuncTickFormatter.from_py_func missing return value and update examples

- [x] issues: fixes #5404 
- [x] tests added / passed

### CODE REVIEW

File: bokeh/models/tests/test_formatters.py [hunk 1]
```diff
@@ -48,33 +50,33 @@ def has_non_Model_keyword_argument(x=10):
...
-};
-return square(tick);
-"""
+    assert formatter.code == dedent("""\
+                                    "use strict";
+                                    var square;
+                                    square = function (x) {
+                                        return x * x;
+                                    };
+                                    return square(tick);
```
Comment (mattpap) [inline bokeh/models/tests/test_formatters.py]: Why so much indentation? This doesn't help readability at all.

Comment (canavandl) [inline bokeh/models/tests/test_formatters.py]: fixed

**Comment by canavandl** [comment]
working formatter in docs:

screen shot 2016-10-26 at 12 54 51 pm

**Comment by bryevdv** [comment]
LGTM

**Comment by bryevdv** [comment]
@canavandl Actually, is it documented somewhere that a python function doesn't take any arguments? If not we should highlight that `tick` is just available (compare it to a module scope var perhaps)

**Comment by canavandl** [comment]
clarified docstrings/user guide

**Comment by bryevdv** [comment]
Perfect, will merge when green

**Pull request merged**